soc/arm/nxp_kinetis/k2x/soc.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2014-2015 Wind River Systems, Inc.
+ * Copyright (c) 2016, Freescale Semiconductor, Inc.
+ * Copyright (c) 2018 Prevas A/S
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for fsl_frdm_k22f platform
+ *
+ * This module provides routines to initialize and support board-level
+ * hardware for the fsl_frdm_k22f platform.
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <uart.h>
+#include <linker/sections.h>
+#include <fsl_common.h>
+#include <fsl_clock.h>
+#include <arch/cpu.h>
+#include <cortex_m/exc.h>
+
+#define PLLFLLSEL_MCGFLLCLK	(0)
+#define PLLFLLSEL_MCGPLLCLK	(1)
+#define PLLFLLSEL_IRC48MHZ	(3)
+
+#define ER32KSEL_OSC32KCLK	(0)
+#define ER32KSEL_RTC		(2)
+#define ER32KSEL_LPO1KHZ	(3)
+
+#define TIMESRC_OSCERCLK        (2)
+
+static const osc_config_t oscConfig = {
+	.freq = CONFIG_OSC_XTAL0_FREQ,
+	.capLoad = 0,
+
+#if defined(CONFIG_OSC_EXTERNAL)
+	.workMode = kOSC_ModeExt,
+#elif defined(CONFIG_OSC_LOW_POWER)
+	.workMode = kOSC_ModeOscLowPower,
+#elif defined(CONFIG_OSC_HIGH_GAIN)
+	.workMode = kOSC_ModeOscHighGain,
+#else
+#error "An oscillator mode must be defined"
+#endif
+
+	.oscerConfig = {
+		.enableMode = 0U, /* Disable external reference clock */
+		.erclkDiv = 0U,
+	},
+};
+
+static const mcg_pll_config_t pll0Config = {
+	.enableMode = 0U,
+	.prdiv = CONFIG_MCG_PRDIV0,
+	.vdiv = CONFIG_MCG_VDIV0,
+};
+
+static const sim_clock_config_t simConfig = {
+	.pllFllSel = PLLFLLSEL_MCGPLLCLK, /* PLLFLLSEL select PLL. */
+	.er32kSrc = ER32KSEL_RTC,         /* ERCLK32K selection, use RTC. */
+	.clkdiv1 = SIM_CLKDIV1_OUTDIV1(CONFIG_K22_CORE_CLOCK_DIVIDER - 1) |
+		   SIM_CLKDIV1_OUTDIV2(CONFIG_K22_BUS_CLOCK_DIVIDER - 1) |
+		   SIM_CLKDIV1_OUTDIV3(CONFIG_K22_FLEXBUS_CLOCK_DIVIDER - 1) |
+		   SIM_CLKDIV1_OUTDIV4(CONFIG_K22_FLASH_CLOCK_DIVIDER - 1),
+};
+
+/**
+ *
+ * @brief Initialize the system clock
+ *
+ * This routine will configure the multipurpose clock generator (MCG) to
+ * set up the system clock.
+ * The MCG has nine possible modes, including Stop mode.  This routine assumes
+ * that the current MCG mode is FLL Engaged Internal (FEI), as from reset.
+ * It transitions through the FLL Bypassed External (FBE) and
+ * PLL Bypassed External (PBE) modes to get to the desired
+ * PLL Engaged External (PEE) mode and generate the maximum 120 MHz system
+ * clock.
+ *
+ * @return N/A
+ *
+ */
+static ALWAYS_INLINE void clkInit(void)
+{
+	CLOCK_SetSimSafeDivs();
+
+	CLOCK_InitOsc0(&oscConfig);
+	CLOCK_SetXtal0Freq(CONFIG_OSC_XTAL0_FREQ);
+
+
+	CLOCK_SetInternalRefClkConfig(kMCG_IrclkEnable, kMCG_IrcSlow,
+				      CONFIG_MCG_FCRDIV);
+
+	/* Configure FLL external reference divider (FRDIV). */
+	/*CLOCK_CONFIG_SetFllExtRefDiv(0);*/
+	MCG->C1 = ((MCG->C1 & ~MCG_C1_FRDIV_MASK) | MCG_C1_FRDIV(0x0U));
```

Comment:
Why did you comment out the CLOCK_ call and modify the MCG register directly?

Reply:
This is leftover from the original PR created by prevas I believe, will fix it